scantool/diag_os_unix.c

```diff
@@ -416,68 +413,12 @@ diag_os_ipending(void) {
 //this is called from most diag_l0_* devices; calling more than once
 //will harm nothing. There is no "opposite" function of this, to
 //reset normal priority.
+//
+//Removed, should be done from a separate process for security reasons.
 int
 diag_os_sched(void) {
-	static int os_sched_done=0;
-	int rv=0;
-
-	if (os_sched_done) {
-		return 0;
-	}
-
-#if defined(_POSIX_PRIORITY_SCHEDULING) && (SEL_SCHED==S_POSIX || SEL_SCHED==S_LINUX || SEL_SCHED==S_AUTO)
-#include <sched.h>
-	/*
-	 * Check privileges
-	 */
-	if (getuid() != 0) {
-		static int suser_warned;
-		if (suser_warned == 0) {
-			suser_warned = 1;
-			fprintf(stderr,
-				FLFMT "WARNING: Not running as superuser; "
-				"things may not work correctly\n", FL);
-		}
-	}
-#if defined(__linux__) && (SEL_SCHED==S_LINUX || SEL_SCHED==S_AUTO)
-	{
-		int r=0;
-		struct sched_param p;
-
-		/* Set real time UNIX scheduling */
-		p.sched_priority = 1;
-		if ( sched_setscheduler(getpid(), SCHED_FIFO, &p) < 0) {
-			fprintf(stderr, FLFMT "sched_setscheduler failed: %s.\n",
-				FL, strerror(errno));
-			r = -1;
-		}
-		rv=r;
-	}
-#else
-	/*
-	+* If we're not running on Linux, we're not sure if what is
-	+* being done is remotely applicable for our flavor of POSIX
-	+* priority scheduling.
-	+* For example, you set the scheduling priority to 1.  Ouch.
```

Comment:
And this comment actually notes the problem.

Reply:
right - but it just does nothing on other platforms. I would keep these code paths in there, they do no harm. The linux implementation could then have a call to sched_get_priority_max() instead of the hardcoded value.

Reply:
* What is the point of trying to set SCHED_FIFO if you agree that scantool should not be a privileged process?

* sched_get_priority_max is POSIX, it is not Linux-specific